lib/migrate_assets_to_asset_manager.rb

```diff
@@ -63,9 +63,9 @@ def legacy_last_modified
     end
 
     def legacy_etag
-      etag = Digest::MD5.hexdigest(read)
-      rewind
-      etag
+      mtime_hex = legacy_last_modified.to_i.to_s(16)
+      file_size_hex = size.to_s(16)
+      [mtime_hex, file_size_hex].join('-')
```

Comment:
Minor:

> I've copied the tests for this behaviour from the equivalent behaviour
added in Asset Manager[1]. I've implemented it without referring to the
implementation in Asset Manager.

It feels a bit perverse to copy the tests, but not the implementation, from Asset Manager. Do you think there's something wrong with the implementation in Asset Manager? Why wouldn't we just copy the implementation?

Reply:
I've just discovered that you ended up copying the implementation from Asset Manager in the next commit, so my previous comment can be disregarded.

Personally, I'd prefer to merge the two commits. If you decide not to do that, I'd be inclined to explain why you haven't copied the implementation in the note for this commit.

Reply:
I've squashed the last two commits.